source/nodejs/tests/ui-tests/src/ui-tests.test.ts

```diff
@@ -8,12 +8,17 @@ describe("Mock function", function() {
     let driver: Webdriver.WebDriver;
     let testUtils: TestUtils.TestUtils;
 
+    // Timeout of 10 minutes for the dev server to start up in the CI jobs, the dev-server
+    // usually takes between 1 to 2 minutes but we have no way to determine when the server
+    // is ready to run tests. This issues is being tracked in issue #6716
+    const timeoutForServerStartupInCIBuild: number = 600000;
```

Comment:
nit: not a blocker for this PR.  it will make the test not very stable.  You may use the same way as [react-native-windows](https://github.com/microsoft/react-native-windows/blob/a58bcdcf643b050e5ad03b42cb2e8f159f997a55/.ado/templates/react-native-init.yml#L48)